pkg/cmd/backuprestore/backupserver.go

```diff
@@ -61,7 +86,49 @@ func (b *backupServer) Run(ctx context.Context) error {
 	defer cancel()
 	signal.NotifyContext(cCtx, shutdownSignals...)
 
+	if b.enabled {
+		schedule, err := cron.ParseStandard(b.schedule)
+		if err != nil {
+			pErr := fmt.Errorf("error parsing backup schedule %v: %w", b.schedule, err)
+			klog.Error(pErr)
+			return pErr
+		}
+
+		err = b.scheduleBackup(cCtx, schedule)
+		if err != nil {
+			return err
+		}
+	}
+
 	<-ctx.Done()
+	return nil
+}
+
+func (b *backupServer) scheduleBackup(ctx context.Context, schedule cron.Schedule) error {
+	ticker := time.NewTicker(time.Until(schedule.Next(time.Now())))
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			err := b.runBackup()
+			if err != nil {
+				return err
```

Comment:
should this return an error or just log? Because returning here will exit the entire sidecar container

Reply:
+1, totally agree

Reply:
maybe also think about this for a follow-up task: how do we surface any errors, do we emit a metric, alerting? Do we put something into the status field of the CRD.

Reply:
I think we need both :)

Reply:
ok .. so how is the alert should be firing ?

Reply:
please create a spike ticket, we can discuss this there.